src/plugins/enzymes/src/FindEnzymesTask.cpp

```diff
@@ -351,7 +351,15 @@ Task *FindEnzymesAutoAnnotationUpdater::createAutoAnnotationsUpdateTask(const Au
 
     U2SequenceObject *dnaObj = aa->getSeqObject();
     const U2Region wholeSequenceRegion = U2Region(0, dnaObj->getSequenceLength());
-    cfg.searchRegion = savedSearchRegion.intersect(wholeSequenceRegion);
+    if (cfg.circular) {
+        cfg.searchRegion = savedSearchRegion;
```

Comment:
Use 1 line: 
cfg.searchRegion = U2Region(savedSearchRegion.startPos, qMin(savedSearchRegion.length, wholeSequenceRegion.length);

And add a comment: "In circular mode the region can have an overflow to handle end/start positions correctly"